lib/client.js

```diff
@@ -334,9 +476,17 @@ function putRaw(self, metric, value, parameters, isMetricAggregated) {
         }
 
         if (!isMetricAggregated) {
```

Comment:
This is why I'm really not a fan of the boolean argument flags.

The code becomes very unclear to follow, maybe you could refactor into different methods?